x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/installation_button.tsx

```diff
@@ -147,15 +333,97 @@ export function InstallationButton(props: InstallationButtonProps) {
+          <>
+            <EuiCallOut
+              color="warning"
+              iconType="alert"
+              title={i18n.translate(
+                'xpack.fleet.integrations.settings.confirmUpdateModal.conflictCallOut.title',
+                { defaultMessage: 'Some integration policies have conflicts' }
+              )}
+            >
+              <strong>
+                <FormattedMessage
+                  id="xpack.fleet.integrations.settings.confirmUpdateModal.conflictCallOut.integrationPolicyCount"
+                  defaultMessage="{conflictCount, plural, one { # integration policy} other { # integration policies}}"
+                  values={{ conflictCount }}
+                />
+              </strong>{' '}
+              <FormattedMessage
+                id="xpack.fleet.integrations.settings.confirmUpdateModal.conflictCallOut.body"
+                defaultMessage="{conflictCount, plural, one { has} other { have}} conflicts and will not be upgraded automatically.
+                  You can manually resolve these conflicts via agent policy settings in Fleet after performing this upgrade."
+                values={{ conflictCount }}
+              />
+            </EuiCallOut>
+
+            <EuiSpacer size="l" />
+          </>
+        ) : null}
+        <FormattedMessage
+          id="xpack.fleet.integrations.settings.confirmUpdateModal.body"
+          defaultMessage="This action will deploy updates to all agents which use these policies.
+          Fleet has detected that {packagePolicyCountText} {packagePolicyCount, plural, one { is} other { are}} ready to be upgraded
+          and {packagePolicyCount, plural, one { is} other { are}} already in use by {agentCountText}."
+          values={{
+            packagePolicyCount,
+            packagePolicyCountText: (
+              <strong>
+                <FormattedMessage
+                  id="xpack.fleet.integrations.confirmUpdateModal.body.policyCount"
+                  defaultMessage="{packagePolicyCount, plural, one {# integration policy} other {# integration policies}}"
+                  values={{ packagePolicyCount }}
+                />
+              </strong>
+            ),
+            agentCountText: (
+              <strong>
+                <FormattedMessage
+                  id="xpack.fleet.integrations.confirmUpdateModal.body.agentCount"
+                  defaultMessage="{agentCount, plural, one {# agent} other {# agents}}"
+                  values={{ agentCount }}
+                />
+              </strong>
+            ),
+          }}
+        />
+      </>
+    </EuiConfirmModal>
+  );
+
   return hasWriteCapabilites ? (
     <Fragment>
       {isUpdate ? updateButton : showUninstallButton ? uninstallButton : installButton}
-      {isModalVisible && (isInstalled ? uninstallModal : installModal)}
+      {isInstallModalVisible && (isInstalled ? uninstallModal : installModal)}
+      {isUpdateModalVisible && updateModal}
```

Comment:
For unit testing purposes, it'd be nice if we could break up this component into:
- A "switch" component that is only responsible for rendering the scenario (eg. install, update, or uninstall)
- A component for each scenario that can be tested separately from the others (eg. `<InstallButtonAndModal />`, `<UninstallButton />`, `<UpdateButtonAndModal />`)

Unit testing this now is really hard due to all of the interrelated dependencies and hooks being used, but if we could split it up there'd be much less to reason about and mock for each component.

If this is feasible, I think we should go ahead and add units for these smaller components in this PR.